<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Provides R users with direct access to genomic and clinical data from the cBioPortal web resource via user-friendly
    functions that wrap cBioPortals existing API endpoints &lt;https://www.cbioportal.org/api/swagger-ui/index.html&gt;. Users can browse and query genomic data on mutations,
    copy number alterations and fusions, as well as data on tumor mutational burden (TMB),
    microsatellite instability status (MSI), FACETS' and select clinical data points (depending on the study). 
    See &lt;https://www.cbioportal.org/&gt; and Gao et al., (2013) &lt;doi:10.1126/scisignal.2004088&gt; for more information on the cBioPortal web resource.">
<title>Browse and Query Clinical and Genomic Data from cBioPortal • cbioportalR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Browse and Query Clinical and Genomic Data from cBioPortal">
<meta property="og:description" content="Provides R users with direct access to genomic and clinical data from the cBioPortal web resource via user-friendly
    functions that wrap cBioPortals existing API endpoints &lt;https://www.cbioportal.org/api/swagger-ui/index.html&gt;. Users can browse and query genomic data on mutations,
    copy number alterations and fusions, as well as data on tumor mutational burden (TMB),
    microsatellite instability status (MSI), FACETS' and select clinical data points (depending on the study). 
    See &lt;https://www.cbioportal.org/&gt; and Gao et al., (2013) &lt;doi:10.1126/scisignal.2004088&gt; for more information on the cBioPortal web resource.">
<meta property="og:image" content="https://karissawhiting.github.io/cbioportalR/logo.png">
<meta name="robots" content="noindex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-ZFP6P2RCRE"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-ZFP6P2RCRE');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">cbioportalR</a>

    <small class="nav-text text-default me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.0.1.9002</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/overview-of-workflow.html">Overview of Data Retrieval Workflow</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/karissawhiting/cbioportalR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9">
<p><br></p>
<div class="section level2">
<h2 id="cbioportalr-">cbioportalR <a href="https://www.karissawhiting.com/cbioportalR/" class="external-link"><img src="reference/figures/logo.png" align="right" height="180" style="align:right; height:180px;"></a>
<a class="anchor" aria-label="anchor" href="#cbioportalr-"></a>
</h2>
<p>{cbioportalR} allows you to access <a href="https://www.cbioportal.org/" class="external-link">cBioPortal’s</a> genomic and clinical data sets directly through R. The package wraps cBioPortal’s API endpoints so R users can easily leverage the <a href="https://www.cbioportal.org/api/swagger-ui/index.html" class="external-link">existing API</a> to access genomic data on mutations, copy number alterations and fusions as well as data on tumor mutational burden (TMB), microsatellite instability status (MSI) and select clinical data points (depending on the study).</p>
<p>This package was created to work with both the public <a href="https://www.cbioportal.org/" class="external-link">cBioPortal website</a>, as well as private institutional cBioPortal instances (e.g. MSKCC, GENIE) with appropriate credentials and <a href="#authentication">authentication</a>.</p>
<p>This package is compatible with cBioPortal v5.0, but is subject to change as <a href="https://github.com/cBioPortal/cbioportal/releases" class="external-link">cBioPortal updates are released</a>. For more information on cBioPortal, see the following publications:</p>
<ul>
<li><a href="https://pubmed.ncbi.nlm.nih.gov/23550210/" class="external-link">Gao et al. Sci. Signal. 2013</a></li>
<li><a href="https://aacrjournals.org/cancerdiscovery/article/2/5/401/3246/The-cBio-Cancer-Genomics-Portal-An-Open-Platform" class="external-link">Cerami et al. Cancer Discov. 2012</a></li>
</ul>
<p>For full documentation on the cBioPortal API, please see the following links:</p>
<ul>
<li><a href="https://docs.cbioportal.org/web-api-and-clients/" class="external-link">cBioPortal API and API Clients documentation</a></li>
<li><a href="https://www.cbioportal.org/api/swagger-ui/index.html" class="external-link">Full reference documentation for API</a></li>
</ul>
<p><em>Note: If you are a MSK researcher working on IMPACT data, you should connect to MSK’s cBioPortal instance to get the most up to date IMPACT data, and you must follow the MSK-IMPACT publication guidelines when using this data</em></p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install {cbioportalR} with the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"cbioportalR"</span><span class="op">)</span></span></code></pre></div>
<p>Install the development version of {cbioportalR} with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"karissawhiting/cbioportalR"</span><span class="op">)</span></span></code></pre></div>
<p>Load the package:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/karissawhiting/cbioportalR" class="external-link">cbioportalR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="authentication">Authentication<a class="anchor" aria-label="anchor" href="#authentication"></a>
</h2>
<p>If you are using the public domain <a href="https://www.cbioportal.org/" class="external-link uri">https://www.cbioportal.org/</a>, you don’t need a token to start pulling data. If you are using a private instance of cBioPortal (like MSKCC’s institutional database), you will need to acquire a token and save it to your <code>.Renviron</code> file (or wherever you store credentials). Simply log in to your institution’s cBioPortal site, acquire a token (usually through the ‘Data Access Token’ link in your username menu in the upper right) and save it in your <code>.Renviron</code> file. This will save the token as an environmental variable so you don’t have to hard code the secret key in your scripts.</p>
<p><em>Tip: The following {usethis} function can easily find and open the <code>.Renviron</code> for you:</em></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">usethis</span><span class="fu">::</span><span class="fu">edit_r_environ</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Paste the token you were given (using the format below) in the .Renviron file and save the file changes. <em>After saving you should restart your R session to ensure the token is saved and recognized.</em></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">CBIOPORTAL_TOKEN</span> <span class="op">=</span> <span class="st">'YOUR_TOKEN'</span></span></code></pre></div>
<p>You can test that your token was saved using:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_cbioportal_token.html">get_cbioportal_token</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>For every new R session, you need to set your database URL. The <code><a href="reference/set_cbioportal_db.html">set_cbioportal_db()</a></code> function will set an environmental variable for your session that tells the package which database to point to for all API calls. You can set it to point to the public database with <code>db = 'www.cbioportal.org'</code> or <code>db = 'public'</code>. If using a private database you will pass your institutions cBioPortal URL as <code>db</code>. This function will both set your URL and check the connection.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/set_cbioportal_db.html">set_cbioportal_db</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"public"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ You are successfully connected!</span></span>
<span><span class="co">#&gt; ✔ base_url for this R session is now set to "www.cbioportal.org/api"</span></span></code></pre></div>
<p>You are now set up for the remainder of your session! API calls depend on your internet connection and possibly a VPN connection so you can use the following to check your connection at any time throughout your session:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/test_cbioportal_db.html">test_cbioportal_db</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ✔ You are successfully connected!</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="cbioportal-data-model">cBioPortal Data Model<a class="anchor" aria-label="anchor" href="#cbioportal-data-model"></a>
</h2>
<p>There are many ways to identify and pull data (e.g. by study ID, by sample ID, by molecular profile ID). Having an understanding of how data is organized in cBioPortal will help you determine which functions you need. The figure below outlines the general data schema for cBioPortal and which functions access which levels of the schema:</p>
<p><img src="reference/figures/readme-cbp-diagram.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p>To see available studies in your database you can use:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/available_studies.html">available_studies</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 10 × 13</span></span>
<span><span class="co">#&gt;    studyId   name  descr…¹ publi…² groups status impor…³ allSa…⁴ readP…⁵ cance…⁶</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;   &lt;lgl&gt;   &lt;chr&gt;   &lt;int&gt; &lt;chr&gt;     &lt;int&gt; &lt;lgl&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt;  1 acc_tcga  Adre… "TCGA … TRUE    "PUBL…      0 2022-0…      92 TRUE    acc    </span></span>
<span><span class="co">#&gt;  2 bcc_unig… Basa… "Whole… TRUE    "PUBL…      0 2022-0…     293 TRUE    bcc    </span></span>
<span><span class="co">#&gt;  3 ampca_bc… Ampu… "Exome… TRUE    "PUBL…      0 2022-0…     160 TRUE    ampca  </span></span>
<span><span class="co">#&gt;  4 blca_dfa… Blad… "Whole… TRUE    "PUBL…      0 2022-0…      50 TRUE    blca   </span></span>
<span><span class="co">#&gt;  5 blca_msk… Blad… "Compr… TRUE    "PUBL…      0 2022-0…      97 TRUE    blca   </span></span>
<span><span class="co">#&gt;  6 blca_bgi  Blad… "Whole… TRUE    "PUBL…      0 2022-0…      99 TRUE    blca   </span></span>
<span><span class="co">#&gt;  7 blca_msk… Blad… "Genom… TRUE    "PUBL…      0 2022-0…     109 TRUE    blca   </span></span>
<span><span class="co">#&gt;  8 all_stju… Hypo… "Whole… TRUE    ""          0 2022-0…      44 TRUE    myeloid</span></span>
<span><span class="co">#&gt;  9 acyc_fmi… Aden… "Targe… TRUE    "ACYC…      0 2022-0…      28 TRUE    acyc   </span></span>
<span><span class="co">#&gt; 10 acyc_san… Aden… "Whole… TRUE    "ACYC…      0 2022-0…      24 TRUE    acyc   </span></span>
<span><span class="co">#&gt; # … with 3 more variables: referenceGenome &lt;chr&gt;, pmid &lt;chr&gt;, citation &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   and abbreviated variable names ¹​description, ²​publicStudy, ³​importDate,</span></span>
<span><span class="co">#&gt; #   ⁴​allSampleCount, ⁵​readPermission, ⁶​cancerTypeId</span></span></code></pre></div>
<p>To view study metadata on a particular study you can use:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_study_info.html">get_study_info</a></span><span class="op">(</span><span class="st">"acc_tcga"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;                             [,1]                                                                                                                                                                                             </span></span>
<span><span class="co">#&gt; name                        "Adrenocortical Carcinoma (TCGA, Firehose Legacy)"                                                                                                                                               </span></span>
<span><span class="co">#&gt; description                 "TCGA Adrenocortical Carcinoma. Source data from &lt;A HREF=\"http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/ACC/20160128/\"&gt;GDAC Firehose&lt;/A&gt;. Previously known as TCGA Provisional."</span></span>
<span><span class="co">#&gt; publicStudy                 "TRUE"                                                                                                                                                                                           </span></span>
<span><span class="co">#&gt; groups                      "PUBLIC"                                                                                                                                                                                         </span></span>
<span><span class="co">#&gt; status                      "0"                                                                                                                                                                                              </span></span>
<span><span class="co">#&gt; importDate                  "2022-03-04 17:47:56"                                                                                                                                                                            </span></span>
<span><span class="co">#&gt; allSampleCount              "92"                                                                                                                                                                                             </span></span>
<span><span class="co">#&gt; sequencedSampleCount        "90"                                                                                                                                                                                             </span></span>
<span><span class="co">#&gt; cnaSampleCount              "90"                                                                                                                                                                                             </span></span>
<span><span class="co">#&gt; mrnaRnaSeqSampleCount       "0"                                                                                                                                                                                              </span></span>
<span><span class="co">#&gt; mrnaRnaSeqV2SampleCount     "79"                                                                                                                                                                                             </span></span>
<span><span class="co">#&gt; mrnaMicroarraySampleCount   "0"                                                                                                                                                                                              </span></span>
<span><span class="co">#&gt; miRnaSampleCount            "0"                                                                                                                                                                                              </span></span>
<span><span class="co">#&gt; methylationHm27SampleCount  "0"                                                                                                                                                                                              </span></span>
<span><span class="co">#&gt; rppaSampleCount             "46"                                                                                                                                                                                             </span></span>
<span><span class="co">#&gt; massSpectrometrySampleCount "0"                                                                                                                                                                                              </span></span>
<span><span class="co">#&gt; completeSampleCount         "75"                                                                                                                                                                                             </span></span>
<span><span class="co">#&gt; readPermission              "TRUE"                                                                                                                                                                                           </span></span>
<span><span class="co">#&gt; studyId                     "acc_tcga"                                                                                                                                                                                       </span></span>
<span><span class="co">#&gt; cancerTypeId                "acc"                                                                                                                                                                                            </span></span>
<span><span class="co">#&gt; cancerType.name             "Adrenocortical Carcinoma"                                                                                                                                                                       </span></span>
<span><span class="co">#&gt; cancerType.dedicatedColor   "Purple"                                                                                                                                                                                         </span></span>
<span><span class="co">#&gt; cancerType.shortName        "ACC"                                                                                                                                                                                            </span></span>
<span><span class="co">#&gt; cancerType.parent           "adrenal_gland"                                                                                                                                                                                  </span></span>
<span><span class="co">#&gt; cancerType.cancerTypeId     "acc"                                                                                                                                                                                            </span></span>
<span><span class="co">#&gt; referenceGenome             "hg19"</span></span></code></pre></div>
<p>To pull all genomic data for a particular study you can use:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_genetics_by_study.html">get_genetics_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"acc_tcga"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ℹ Returning all data for the "acc_tcga_mutations" molecular profile in the "acc_tcga" study</span></span>
<span><span class="co">#&gt; ℹ Returning all data for the "acc_tcga_gistic" molecular profile in the "acc_tcga" study</span></span>
<span><span class="co">#&gt; ! No "structural_variant" data returned. Error:  No molecular profile for `data_type = fusion` found in "acc_tcga".  See `available_profiles('acc_tcga')`</span></span></code></pre></div>
<p>As a result, you will get a list of data frames with mutation and CNA data respectively. The function will also try to pull fusion (structural variant) data, but there is no fusion data available for this study, as indicated by the function message.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">df</span><span class="op">$</span><span class="va">mutation</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 33</span></span>
<span><span class="co">#&gt;   hugoG…¹ entre…² uniqu…³ uniqu…⁴ molec…⁵ sampl…⁶ patie…⁷ studyId center mutat…⁸</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;     &lt;int&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; 1 KRT8       3856 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… broad… Somatic</span></span>
<span><span class="co">#&gt; 2 LCE1B    353132 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… hgsc.… Somatic</span></span>
<span><span class="co">#&gt; 3 SLC9C2   284525 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… broad… Somatic</span></span>
<span><span class="co">#&gt; 4 DNAH14   127602 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… broad… Somatic</span></span>
<span><span class="co">#&gt; 5 OPN4      94233 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… hgsc.… Somatic</span></span>
<span><span class="co">#&gt; 6 DNAJC4     3338 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… hgsc.… Somatic</span></span>
<span><span class="co">#&gt; # … with 23 more variables: validationStatus &lt;chr&gt;, tumorAltCount &lt;int&gt;,</span></span>
<span><span class="co">#&gt; #   tumorRefCount &lt;int&gt;, normalAltCount &lt;int&gt;, normalRefCount &lt;int&gt;,</span></span>
<span><span class="co">#&gt; #   startPosition &lt;int&gt;, endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   proteinChange &lt;chr&gt;, mutationType &lt;chr&gt;, functionalImpactScore &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   fisValue &lt;dbl&gt;, linkXvar &lt;chr&gt;, linkPdb &lt;chr&gt;, linkMsa &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, keyword &lt;chr&gt;, chr &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   variantAllele &lt;chr&gt;, refseqMrnaId &lt;chr&gt;, proteinPosStart &lt;int&gt;, …</span></span>
<span></span>
<span><span class="va">df</span><span class="op">$</span><span class="va">cna</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 9</span></span>
<span><span class="co">#&gt;   hugoGeneSymbol entre…¹ uniqu…² uniqu…³ molec…⁴ sampl…⁵ patie…⁶ studyId alter…⁷</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;            &lt;int&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;     &lt;int&gt;</span></span>
<span><span class="co">#&gt; 1 MEOX1             4222 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc…       2</span></span>
<span><span class="co">#&gt; 2 NUFIP2           57532 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc…       2</span></span>
<span><span class="co">#&gt; 3 OSBPL7          114881 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc…       2</span></span>
<span><span class="co">#&gt; 4 TP53I13          90313 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc…       2</span></span>
<span><span class="co">#&gt; 5 TAOK1            57551 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc…       2</span></span>
<span><span class="co">#&gt; 6 SPOP              8405 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc…       2</span></span>
<span><span class="co">#&gt; # … with abbreviated variable names ¹​entrezGeneId, ²​uniqueSampleKey,</span></span>
<span><span class="co">#&gt; #   ³​uniquePatientKey, ⁴​molecularProfileId, ⁵​sampleId, ⁶​patientId, ⁷​alteration</span></span></code></pre></div>
<p>You can also pull data by specific sample IDs but the API requires a bit more information from you (unlike pulling by study ID which returns everything available for that study). This can be useful when working within a very large database or working across samples housed in multiple different studies. When querying by <code>sample_id</code> you must also specify the corresponding <code>study_id</code> in which the samples are housed. When these pieces of information are not provided, {cbioportalR} makes an informed guess based on your connection and will throw an informative message to clarify exactly what is being queried. In the example below, the function defaults to the public version of the IMPACT database (<code>study_id = "msk_impact_2017"</code>).</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/available_samples.html">available_samples</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"acc_tcga"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">sampleId</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mutations</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_mutations_by_sample.html">get_mutations_by_sample</a></span><span class="op">(</span>sample_id <span class="op">=</span>  <span class="va">samples</span><span class="op">)</span></span>
<span><span class="co">#&gt; The following parameters were used in query:</span></span>
<span><span class="co">#&gt; Study ID: "msk_impact_2017"</span></span>
<span><span class="co">#&gt; Molecular Profile ID: "msk_impact_2017_mutations"</span></span>
<span><span class="co">#&gt; Genes: "All available genes"</span></span>
<span></span>
<span><span class="co"># no results returned because these samples are not in this study</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">mutations</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span></span>
<span><span class="co">#&gt; [1] TRUE</span></span></code></pre></div>
<p>No results were returned because the samples are not stored in this study. When we specify the correct study (<code>study_id = "acc_tcga"</code>), we get accurate results. You can check which samples are stored in a study using <code>available_samples(study_id = "acc_tcga")</code>.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mutations</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/get_mutations_by_sample.html">get_mutations_by_sample</a></span><span class="op">(</span>sample_id <span class="op">=</span>  <span class="va">samples</span>,</span>
<span>                                    study_id <span class="op">=</span> <span class="st">"acc_tcga"</span><span class="op">)</span></span>
<span><span class="co">#&gt; The following parameters were used in query:</span></span>
<span><span class="co">#&gt; Study ID: "acc_tcga"</span></span>
<span><span class="co">#&gt; Molecular Profile ID: "acc_tcga_mutations"</span></span>
<span><span class="co">#&gt; Genes: "All available genes"</span></span>
<span></span>
<span><span class="va">mutations</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 33</span></span>
<span><span class="co">#&gt;   hugoG…¹ entre…² uniqu…³ uniqu…⁴ molec…⁵ sampl…⁶ patie…⁷ studyId center mutat…⁸</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;     &lt;int&gt; &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; 1 KRT8       3856 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… broad… Somatic</span></span>
<span><span class="co">#&gt; 2 LCE1B    353132 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… hgsc.… Somatic</span></span>
<span><span class="co">#&gt; 3 SLC9C2   284525 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… broad… Somatic</span></span>
<span><span class="co">#&gt; 4 DNAH14   127602 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… broad… Somatic</span></span>
<span><span class="co">#&gt; 5 OPN4      94233 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… hgsc.… Somatic</span></span>
<span><span class="co">#&gt; 6 DNAJC4     3338 VENHQS… VENHQS… acc_tc… TCGA-O… TCGA-O… acc_tc… hgsc.… Somatic</span></span>
<span><span class="co">#&gt; # … with 23 more variables: validationStatus &lt;chr&gt;, tumorAltCount &lt;int&gt;,</span></span>
<span><span class="co">#&gt; #   tumorRefCount &lt;int&gt;, normalAltCount &lt;int&gt;, normalRefCount &lt;int&gt;,</span></span>
<span><span class="co">#&gt; #   startPosition &lt;int&gt;, endPosition &lt;int&gt;, referenceAllele &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   proteinChange &lt;chr&gt;, mutationType &lt;chr&gt;, functionalImpactScore &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   fisValue &lt;dbl&gt;, linkXvar &lt;chr&gt;, linkPdb &lt;chr&gt;, linkMsa &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   ncbiBuild &lt;chr&gt;, variantType &lt;chr&gt;, keyword &lt;chr&gt;, chr &lt;chr&gt;,</span></span>
<span><span class="co">#&gt; #   variantAllele &lt;chr&gt;, refseqMrnaId &lt;chr&gt;, proteinPosStart &lt;int&gt;, …</span></span></code></pre></div>
<p>Lastly, you can also pull clinical data or sample metadata (e.g. tumor sample site) by study ID, sample ID or patient ID. To see what data is available, you can use:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/available_clinical_attributes.html">available_clinical_attributes</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"acc_tcga"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 × 7</span></span>
<span><span class="co">#&gt;   displayName                    descr…¹ datat…² patie…³ prior…⁴ clini…⁵ studyId</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;                          &lt;chr&gt;   &lt;chr&gt;   &lt;lgl&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;  </span></span>
<span><span class="co">#&gt; 1 Diagnosis Age                  Age at… NUMBER  TRUE    1       AGE     acc_tc…</span></span>
<span><span class="co">#&gt; 2 Neoplasm Disease Stage Americ… The ex… STRING  TRUE    1       AJCC_P… acc_tc…</span></span>
<span><span class="co">#&gt; 3 American Joint Committee on C… The ve… STRING  TRUE    1       AJCC_S… acc_tc…</span></span>
<span><span class="co">#&gt; 4 Atypical Mitotic Figures       Atypic… STRING  TRUE    1       ATYPIC… acc_tc…</span></span>
<span><span class="co">#&gt; 5 Cancer Type                    Cancer… STRING  FALSE   1       CANCER… acc_tc…</span></span>
<span><span class="co">#&gt; 6 Cancer Type Detailed           Cancer… STRING  FALSE   1       CANCER… acc_tc…</span></span>
<span><span class="co">#&gt; # … with abbreviated variable names ¹​description, ²​datatype, ³​patientAttribute,</span></span>
<span><span class="co">#&gt; #   ⁴​priority, ⁵​clinicalAttributeId</span></span></code></pre></div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_clinical_by_study.html">get_clinical_by_study</a></span><span class="op">(</span><span class="st">"acc_tcga"</span><span class="op">)</span></span>
<span><span class="co">#&gt; ! Sample Level Clinical Data: No `clinical_attribute` passed. Defaulting to returning all clinical attributes in "acc_tcga" study</span></span>
<span><span class="co">#&gt; ! Patient Level Clinical Data: No `clinical_attribute` passed. Defaulting to returning all clinical attributes in "acc_tcga" study</span></span>
<span><span class="co">#&gt; # A tibble: 6,292 × 6</span></span>
<span><span class="co">#&gt;    uniquePatientKey             patientId    studyId  clinicalAt…¹ value dataL…²</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                        &lt;chr&gt;        &lt;chr&gt;    &lt;chr&gt;        &lt;chr&gt; &lt;chr&gt;  </span></span>
<span><span class="co">#&gt;  1 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga AGE          58    PATIENT</span></span>
<span><span class="co">#&gt;  2 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga AJCC_PATHOL… Stag… PATIENT</span></span>
<span><span class="co">#&gt;  3 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga ATYPICAL_MI… Atyp… PATIENT</span></span>
<span><span class="co">#&gt;  4 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga CAPSULAR_IN… Inva… PATIENT</span></span>
<span><span class="co">#&gt;  5 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga CLIN_M_STAGE M0    PATIENT</span></span>
<span><span class="co">#&gt;  6 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga CT_SCAN_PRE… [Unk… PATIENT</span></span>
<span><span class="co">#&gt;  7 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga CYTOPLASM_P… Cyto… PATIENT</span></span>
<span><span class="co">#&gt;  8 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga DAYS_TO_INI… 0     PATIENT</span></span>
<span><span class="co">#&gt;  9 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga DFS_MONTHS   24.77 PATIENT</span></span>
<span><span class="co">#&gt; 10 VENHQS1PUi1BNUoxOmFjY190Y2dh TCGA-OR-A5J1 acc_tcga DFS_STATUS   1:Re… PATIENT</span></span>
<span><span class="co">#&gt; # … with 6,282 more rows, and abbreviated variable names ¹​clinicalAttributeId,</span></span>
<span><span class="co">#&gt; #   ²​dataLevel</span></span></code></pre></div>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/get_clinical_by_sample.html">get_clinical_by_sample</a></span><span class="op">(</span>sample_id <span class="op">=</span> <span class="va">samples</span>, study_id <span class="op">=</span> <span class="st">"acc_tcga"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span>
<span><span class="co">#&gt; ! No `clinical_attribute` passed. Defaulting to returning</span></span>
<span><span class="co">#&gt; all clinical attributes in "acc_tcga" study</span></span>
<span><span class="co">#&gt; # A tibble: 10 × 7</span></span>
<span><span class="co">#&gt;    uniqueSampleKey                 uniqu…¹ sampl…² patie…³ studyId clini…⁴ value</span></span>
<span><span class="co">#&gt;    &lt;chr&gt;                           &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;   &lt;chr&gt;</span></span>
<span><span class="co">#&gt;  1 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… CANCER… Adre…</span></span>
<span><span class="co">#&gt;  2 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… CANCER… Adre…</span></span>
<span><span class="co">#&gt;  3 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… DAYS_T… 4691 </span></span>
<span><span class="co">#&gt;  4 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… FRACTI… 0.05…</span></span>
<span><span class="co">#&gt;  5 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… IS_FFPE NO   </span></span>
<span><span class="co">#&gt;  6 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… MUTATI… 39   </span></span>
<span><span class="co">#&gt;  7 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… OCT_EM… TRUE </span></span>
<span><span class="co">#&gt;  8 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… ONCOTR… ACC  </span></span>
<span><span class="co">#&gt;  9 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… OTHER_… E403…</span></span>
<span><span class="co">#&gt; 10 VENHQS1PUi1BNUoxLTAxOmFjY190Y2… VENHQS… TCGA-O… TCGA-O… acc_tc… PATHOL… TCGA…</span></span>
<span><span class="co">#&gt; # … with abbreviated variable names ¹​uniquePatientKey, ²​sampleId, ³​patientId,</span></span>
<span><span class="co">#&gt; #   ⁴​clinicalAttributeId</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">patients</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/available_patients.html">available_patients</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"acc_tcga"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">pull</span><span class="op">(</span><span class="va">patientId</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="reference/get_clinical_by_patient.html">get_clinical_by_patient</a></span><span class="op">(</span>patient_id <span class="op">=</span> <span class="va">patients</span>, study_id <span class="op">=</span> <span class="st">"acc_tcga"</span>, </span>
<span>                        clinical_attribute <span class="op">=</span> <span class="st">"AGE"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>All functions that pull by study IDs are limited to pulling data from one study at a time. If you need to pull specific samples from multiple studies, you likely want to pull by sample ID (instead of study ID) and supply the function with a dataframe of <code>sample_study_pairs</code> that specify where the function should look for each study. For more information see the <a href="https://www.karissawhiting.com/cbioportalR/articles/overview-of-workflow.html" class="external-link">Overview of Workflow Vignette</a>.</p>
</div>
<div class="section level2">
<h2 id="contributing">Contributing<a class="anchor" aria-label="anchor" href="#contributing"></a>
</h2>
<p>Please note that {cbioportalR} is released with a <a href="https://www.karissawhiting.com/cbioportalR/CODE_OF_CONDUCT.html" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
<p>Thank you to contributors!</p>
<p><a href="https://github.com/arorarshi" class="external-link">@arorarshi</a>, <a href="https://github.com/AxelitoMartin" class="external-link">@AxelitoMartin</a>, <a href="https://github.com/edrill" class="external-link">@edrill</a>, <a href="https://github.com/jalavery" class="external-link">@jalavery</a>, <a href="https://github.com/ddsjoberg" class="external-link">@ddsjoberg</a> <a href="https://github.com/karomanchuk" class="external-link">@karomanchuk</a> <a href="https://github.com/hfuchs5" class="external-link">@hfuchs5</a> <a href="https://github.com/alrein-05" class="external-link">@alrein-05</a></p>
<p>Thank you <a href="https://awenfilms.net/" class="external-link">Isaak Liptzin</a> for the hex sticker!</p>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=cbioportalR" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/karissawhiting/cbioportalR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/karissawhiting/cbioportalR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing cbioportalR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-4683-1868" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/karissawhiting/cbioportalR/actions" class="external-link"><img src="https://github.com/karissawhiting/cbioportalR/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/karissawhiting/cbioportalR?branch=master" class="external-link"><img src="https://codecov.io/gh/karissawhiting/cbioportalR/branch/master/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://CRAN.R-project.org/package=cbioportalR" class="external-link"><img src="https://www.r-pkg.org/badges/version/cbioportalR" alt="CRAN status"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
