<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cbioportalR">
<title>Tutorial: Overview of Data Retrieval Workflow • cbioportalR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Tutorial: Overview of Data Retrieval Workflow">
<meta property="og:description" content="cbioportalR">
<meta name="robots" content="noindex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">cbioportalR</a>

    <small class="nav-text text-default me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">0.1.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/tutorial-workflow-overview.html">Tutorial: Overview of Data Retrieval Workflow</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/karissawhiting/cbioportalR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="tutorial-workflow-overview_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Tutorial: Overview of Data Retrieval Workflow</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/karissawhiting/cbioportalR/blob/HEAD/vignettes/tutorial-workflow-overview.Rmd" class="external-link"><code>vignettes/tutorial-workflow-overview.Rmd</code></a></small>
      <div class="d-none name"><code>tutorial-workflow-overview.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/karissawhiting/cbioportalR" class="external-link">cbioportalR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>We will outline the main data retrieval workflow and functions using a case study based on a public set of data on 105 samples in high risk nonmuscle invasive bladder cancer patients <a href="https://pubmed.ncbi.nlm.nih.gov/28583311/" class="external-link">(Pietzak et al. 2017)</a>.</p>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>Before accessing data you will need to connect to a cBioPortal database and set your base URL for the R session. In this example we will use data from the public cBioPortal database instance (<a href="https://www.cbioportal.org" class="external-link uri">https://www.cbioportal.org</a>). You do not need a token to access this public website. If you are using a private instance of cbioportal (like MSK’s institutional database), you will need to acquire a token and save it to your .Renviron file (see Authentication vignetted for more details -coming soon)</p>
<p><em>Note: If you are a MSK researcher working on IMPACT, you should connect to MSK’s cBioPortal instance to get the most up to date IMPACT data, and you must follow MSK-IMPACT publication guidelines when using the data.</em></p>
<p>To set the database urlfor your current R session use the <code><a href="../reference/set_cbioportal_db.html">set_cbioportal_db()</a></code> function. To set it to the public instance you can either provide the full URL to the function, or just <code>public</code> as a shortcut.</p>
<p>This function will both check your connection to the database and set the database url (<code>www.cbioportal.org/api</code>) as your base url to connect to for all future API calls during your session.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="../reference/set_cbioportal_db.html">set_cbioportal_db</a></span><span class="op">(</span><span class="st">"public"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> You are successfully connected!</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">base_url</span> for this R session is now set to <span style="color: #0000BB;">"www.cbioportal.org/api"</span></span></code></pre></div>
<p>You can use <code>test_cbioportal_db</code> at any time throughout your session to check your connection. This can be helpful when troubleshooting issues with your API calls.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/test_cbioportal_db.html">test_cbioportal_db</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> You are successfully connected!</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get-study-metadata">Get Study Metadata<a class="anchor" aria-label="anchor" href="#get-study-metadata"></a>
</h2>
<p>Now that we are successfully connected, we may want to view all studies available for our chosen database to find the correct <code>study_id</code> corresponding to the data we want to pull. All studies have a unique identifier in the database. You can view all studies available in your database with the following:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_studies.html">available_studies</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">all_studies</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 340 × 13</span></span>
<span class="co">#&gt;    studyId name  description publicStudy pmid  citation groups status importDate</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> prad_m… Pros… <span style="color: #949494;">"</span>Genome-wi… TRUE        2502… Hierony… <span style="color: #949494;">"</span>PUBL…      0 2021-04-2…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> pan_or… Chin… <span style="color: #949494;">"</span>Landscape… TRUE        <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>       <span style="color: #949494;">""</span>          0 2021-09-2…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> crc_ni… Disp… <span style="color: #949494;">"</span>Targeted … TRUE        <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>       <span style="color: #949494;">""</span>          0 2021-10-2…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> acc_tc… Adre… <span style="color: #949494;">"</span>TCGA Adre… TRUE        <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>       <span style="color: #949494;">"</span>PUBL…      0 2021-10-2…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> acyc_s… Aden… <span style="color: #949494;">"</span>Whole exo… TRUE        2377… Stephen… <span style="color: #949494;">"</span>ACYC…      0 2021-10-2…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> all_st… Acut… <span style="color: #949494;">"</span>Whole-gen… TRUE        2777… Zhang e… <span style="color: #949494;">"</span>PUBL…      0 2021-10-2…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> all_ph… Pedi… <span style="color: #949494;">"</span>Whole gen… TRUE        <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>       <span style="color: #949494;">"</span>NCI-…      0 2021-10-2…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> all_st… Acut… <span style="color: #949494;">"</span>Comprehen… TRUE        2573… Anderss… <span style="color: #949494;">"</span>PUBL…      0 2021-11-0…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> acbc_m… Aden… <span style="color: #949494;">"</span>Whole exo… TRUE        2609… Martelo… <span style="color: #949494;">"</span>ACYC…      0 2021-11-0…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> acyc_j… Aden… <span style="color: #949494;">"</span>Whole-gen… TRUE        2686… Rettig … <span style="color: #949494;">"</span>ACYC…      0 2021-11-0…</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 330 more rows, and 4 more variables: allSampleCount &lt;int&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   readPermission &lt;lgl&gt;, cancerTypeId &lt;chr&gt;, referenceGenome &lt;chr&gt;</span></span></code></pre></div>
<p>By inspecting this dataframe, we see the unique <code>study_id</code> we want is <code>"blca_nmibc_2017"</code> To get more information on our study we can do the following:</p>
<p><em>Note: the transpose function <code><a href="https://rdrr.io/r/base/t.html" class="external-link">t()</a></code> just used here to better view results</em></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_studies</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">studyId</span> <span class="op">==</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;                 [,1]                                                                           </span>
<span class="co">#&gt; studyId         "blca_nmibc_2017"                                                              </span>
<span class="co">#&gt; name            "Nonmuscle Invasive Bladder Cancer (MSK Eur Urol 2017)"                        </span>
<span class="co">#&gt; description     "IMPACT sequencing of 105 High Risk Nonmuscle Invasive Bladder Cancer samples."</span>
<span class="co">#&gt; publicStudy     "TRUE"                                                                         </span>
<span class="co">#&gt; pmid            "28583311"                                                                     </span>
<span class="co">#&gt; citation        "Pietzak et al. Eur Urol 2017"                                                 </span>
<span class="co">#&gt; groups          "PUBLIC"                                                                       </span>
<span class="co">#&gt; status          "0"                                                                            </span>
<span class="co">#&gt; importDate      "2021-11-03 00:00:00"                                                          </span>
<span class="co">#&gt; allSampleCount  "105"                                                                          </span>
<span class="co">#&gt; readPermission  "TRUE"                                                                         </span>
<span class="co">#&gt; cancerTypeId    "blca"                                                                         </span>
<span class="co">#&gt; referenceGenome "hg19"</span></code></pre></div>
<p>More in-depth information about the study can be found with <code><a href="../reference/get_study_info.html">get_study_info()</a></code></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_study_info.html">get_study_info</a></span><span class="op">(</span><span class="st">"blca_nmibc_2017"</span><span class="op">)</span><span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span> 
<span class="co">#&gt;                             [,1]                                                                           </span>
<span class="co">#&gt; name                        "Nonmuscle Invasive Bladder Cancer (MSK Eur Urol 2017)"                        </span>
<span class="co">#&gt; description                 "IMPACT sequencing of 105 High Risk Nonmuscle Invasive Bladder Cancer samples."</span>
<span class="co">#&gt; publicStudy                 "TRUE"                                                                         </span>
<span class="co">#&gt; pmid                        "28583311"                                                                     </span>
<span class="co">#&gt; citation                    "Pietzak et al. Eur Urol 2017"                                                 </span>
<span class="co">#&gt; groups                      "PUBLIC"                                                                       </span>
<span class="co">#&gt; status                      "0"                                                                            </span>
<span class="co">#&gt; importDate                  "2021-11-03 00:00:00"                                                          </span>
<span class="co">#&gt; allSampleCount              "105"                                                                          </span>
<span class="co">#&gt; sequencedSampleCount        "105"                                                                          </span>
<span class="co">#&gt; cnaSampleCount              "105"                                                                          </span>
<span class="co">#&gt; mrnaRnaSeqSampleCount       "0"                                                                            </span>
<span class="co">#&gt; mrnaRnaSeqV2SampleCount     "0"                                                                            </span>
<span class="co">#&gt; mrnaMicroarraySampleCount   "0"                                                                            </span>
<span class="co">#&gt; miRnaSampleCount            "0"                                                                            </span>
<span class="co">#&gt; methylationHm27SampleCount  "0"                                                                            </span>
<span class="co">#&gt; rppaSampleCount             "0"                                                                            </span>
<span class="co">#&gt; massSpectrometrySampleCount "0"                                                                            </span>
<span class="co">#&gt; completeSampleCount         "0"                                                                            </span>
<span class="co">#&gt; readPermission              "TRUE"                                                                         </span>
<span class="co">#&gt; studyId                     "blca_nmibc_2017"                                                              </span>
<span class="co">#&gt; cancerTypeId                "blca"                                                                         </span>
<span class="co">#&gt; cancerType.name             "Bladder Urothelial Carcinoma"                                                 </span>
<span class="co">#&gt; cancerType.dedicatedColor   "Yellow"                                                                       </span>
<span class="co">#&gt; cancerType.shortName        "BLCA"                                                                         </span>
<span class="co">#&gt; cancerType.parent           "bladder"                                                                      </span>
<span class="co">#&gt; cancerType.cancerTypeId     "blca"                                                                         </span>
<span class="co">#&gt; referenceGenome             "hg19"</span></code></pre></div>
<p>Lastly, it is important to know what genomic data is available for our study. Not all studies in your database will have data available on all types of genomic information. For example, it is common for studies not to provide data on fusions.</p>
<p>We can check available genomic data with <code><a href="../reference/available_profiles.html">available_profiles()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/available_profiles.html">available_profiles</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 8</span></span>
<span class="co">#&gt;   molecularAlterationT… datatype name  description showProfileInAn… patientLevel</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> COPY_NUMBER_ALTERATI… DISCRETE Puta… Copy Numbe… TRUE             FALSE       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> MUTATION_EXTENDED     MAF      Muta… Mutation d… TRUE             FALSE       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> STRUCTURAL_VARIANT    FUSION   Fusi… Fusions.    TRUE             FALSE       </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 2 more variables: molecularProfileId &lt;chr&gt;, studyId &lt;chr&gt;</span></span></code></pre></div>
<p>Luckily, our study has mutation, copy number alteration and fusion data available. Each of these data types has a unique molecular profile ID. The molecular profile ID usually takes the form of <code>&lt;study_id&gt;_mutations</code>, <code>&lt;study_id&gt;_fusion</code>, <code>&lt;study_id&gt;_cna</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/available_profiles.html">available_profiles</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">molecularProfileId</span><span class="op">)</span>
<span class="co">#&gt; [1] "blca_nmibc_2017_cna"       "blca_nmibc_2017_mutations"</span>
<span class="co">#&gt; [3] "blca_nmibc_2017_fusion"</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pulling-genomic-data">Pulling Genomic Data<a class="anchor" aria-label="anchor" href="#pulling-genomic-data"></a>
</h2>
<p>Now that we have inspected our study and confirmed the genomic data that is available, we will pull the data into our R environment.</p>
<p>Since we are working with only one study, the easiest way to do this is by study_id. We can pull each data type individually:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">mut</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mutation_by_study.html">get_mutation_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_mutations"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>
<span class="va">cna</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_cna_by_study.html">get_cna_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_cna"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>
<span class="va">fus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_fusion_by_study.html">get_fusion_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_fusion"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span></code></pre></div>
<p>Or we can pull all genomic data at the same time with <code><a href="../reference/get_genetics_by_study.html">get_genetics_by_study()</a></code></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_genomic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genetics_by_study.html">get_genetics_by_study</a></span><span class="op">(</span><span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_mutations"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_cna"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_fusion"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://dplyr.tidyverse.org/reference/all_equal.html" class="external-link">all_equal</a></span><span class="op">(</span><span class="va">mut</span>, <span class="va">all_genomic</span><span class="op">$</span><span class="va">mutation</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/all_equal.html" class="external-link">all_equal</a></span><span class="op">(</span><span class="va">cna</span>, <span class="va">all_genomic</span><span class="op">$</span><span class="va">cna</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/all_equal.html" class="external-link">all_equal</a></span><span class="op">(</span><span class="va">fus</span>, <span class="va">all_genomic</span><span class="op">$</span><span class="va">fusion</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#get_panel_by_sample</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pulling-clinical-data">Pulling Clinical Data<a class="anchor" aria-label="anchor" href="#pulling-clinical-data"></a>
</h2>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Karissa Whiting, Axel Martin, Arshi Arora.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
