<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="cbioportalR">
<title>Overview of Data Retrieval Workflow • cbioportalR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Overview of Data Retrieval Workflow">
<meta property="og:description" content="cbioportalR">
<meta property="og:image" content="https://karissawhiting.github.io/cbioportalR/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">cbioportalR</a>

    <small class="nav-text text-default me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">0.2.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/internal-structure.html">Internal Structure &amp; Guiding Principles</a>
    <a class="dropdown-item" href="../articles/overview-of-workflow.html">Overview of Data Retrieval Workflow</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/karissawhiting/cbioportalR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="overview-of-workflow_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Overview of Data Retrieval Workflow</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/karissawhiting/cbioportalR/blob/HEAD/vignettes/overview-of-workflow.Rmd" class="external-link"><code>vignettes/overview-of-workflow.Rmd</code></a></small>
      <div class="d-none name"><code>overview-of-workflow.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/karissawhiting/cbioportalR" class="external-link">cbioportalR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>We will outline the main data retrieval workflow and functions using a case study based on two public sets of data:</p>
<ol style="list-style-type: decimal">
<li>105 samples in high risk nonmuscle invasive bladder cancer patients <a href="https://pubmed.ncbi.nlm.nih.gov/28583311/" class="external-link">(Pietzak et al. 2017)</a>.</li>
<li>18 samples of 18 prostate cancer patients <a href="https://pubmed.ncbi.nlm.nih.gov/31564440/" class="external-link">(Granlund et al. 2020)</a>
</li>
</ol>
</div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<p>Before accessing data you will need to connect to a cBioPortal database and set your base URL for the R session. In this example we will use data from the public cBioPortal database instance (<a href="https://www.cbioportal.org" class="external-link uri">https://www.cbioportal.org</a>). You do not need a token to access this public website. If you are using a private instance of cBioPortal (like MSK’s institutional database), you will need to acquire a token and save it to your .Renviron file.</p>
<p><em>Note: If you are a MSK researcher working on IMPACT, you should connect to MSK’s cBioPortal instance to get the most up to date IMPACT data, and you must follow MSK-IMPACT publication guidelines when using the data.</em></p>
<p>To set the database url for your current R session use the <code><a href="../reference/set_cbioportal_db.html">set_cbioportal_db()</a></code> function. To set it to the public instance you can either provide the full URL to the function, or just <code>public</code> as a shortcut. This function will both check your connection to the database and set the url (<code>www.cbioportal.org/api</code>) as your base url to connect to for all future API calls during your session.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/set_cbioportal_db.html">set_cbioportal_db</a></span><span class="op">(</span><span class="st">"public"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> You are successfully connected!</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">base_url</span> for this R session is now set to <span style="color: #0000BB;">"www.cbioportal.org/api"</span></span></code></pre></div>
<p>You can use <code>test_cbioportal_db</code> at any time throughout your session to check your connection. This can be helpful when troubleshooting issues with your API calls.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/test_cbioportal_db.html">test_cbioportal_db</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BB00;">✔</span> You are successfully connected!</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get-study-metadata">Get Study Metadata<a class="anchor" aria-label="anchor" href="#get-study-metadata"></a>
</h2>
<p>Now that we are successfully connected, we may want to view all studies available for our chosen database to find the correct <code>study_id</code> corresponding to the data we want to pull. All studies have a unique identifier in the database. You can view all studies available in your database with the following:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_studies</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_studies.html">available_studies</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">all_studies</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 342 × 13</span></span>
<span class="co">#&gt;    studyId name  description publicStudy groups status importDate allSampleCount</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> acc_tc… Adre… <span style="color: #949494;">"</span>TCGA Adre… TRUE        <span style="color: #949494;">"</span>PUBL…      0 2022-03-0…             92</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> blca_p… Blad… <span style="color: #949494;">"</span>Whole exo… TRUE        <span style="color: #949494;">""</span>          0 2022-03-0…             34</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> bcc_un… Basa… <span style="color: #949494;">"</span>Whole-exo… TRUE        <span style="color: #949494;">"</span>PUBL…      0 2022-03-0…            293</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> all_st… Acut… <span style="color: #949494;">"</span>Comprehen… TRUE        <span style="color: #949494;">"</span>PUBL…      0 2022-03-0…             93</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ampca_… Ampu… <span style="color: #949494;">"</span>Exome seq… TRUE        <span style="color: #949494;">"</span>PUBL…      0 2022-03-0…            160</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> blca_d… Blad… <span style="color: #949494;">"</span>Whole exo… TRUE        <span style="color: #949494;">"</span>PUBL…      0 2022-03-0…             50</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> blca_m… Blad… <span style="color: #949494;">"</span>Comprehen… TRUE        <span style="color: #949494;">"</span>PUBL…      0 2022-03-0…             97</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> blca_b… Blad… <span style="color: #949494;">"</span>Whole-exo… TRUE        <span style="color: #949494;">"</span>PUBL…      0 2022-03-0…             99</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> blca_m… Blad… <span style="color: #949494;">"</span>Genomic P… TRUE        <span style="color: #949494;">"</span>PUBL…      0 2022-03-0…            109</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> all_st… Hypo… <span style="color: #949494;">"</span>Whole gen… TRUE        <span style="color: #949494;">""</span>          0 2022-03-0…             44</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 332 more rows, and 5 more variables: readPermission &lt;lgl&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   cancerTypeId &lt;chr&gt;, referenceGenome &lt;chr&gt;, pmid &lt;chr&gt;, citation &lt;chr&gt;</span></span></code></pre></div>
<p>By inspecting this dataframe, we see the unique <code>study_id</code> we want is <code>"blca_nmibc_2017"</code> To get more information on our study we can do the following:</p>
<p><em>Note: the transpose function <code><a href="https://rdrr.io/r/base/t.html" class="external-link">t()</a></code> just used here to better view results</em></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_studies</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">studyId</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"blca_nmibc_2017"</span>, <span class="st">"prad_msk_2019"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 13</span></span>
<span class="co">#&gt;   studyId  name  description publicStudy groups status importDate allSampleCount</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> blca_nm… Nonm… IMPACT seq… TRUE        PUBLIC      0 2022-03-0…            105</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> prad_ms… Pros… MSK-IMPACT… TRUE        PUBLIC      0 2022-03-0…             18</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 5 more variables: readPermission &lt;lgl&gt;, cancerTypeId &lt;chr&gt;,</span></span>
<span class="co">#&gt; <span style="color: #949494;">#   referenceGenome &lt;chr&gt;, pmid &lt;chr&gt;, citation &lt;chr&gt;</span></span></code></pre></div>
<p>More in-depth information about the study can be found with <code><a href="../reference/get_study_info.html">get_study_info()</a></code></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_study_info.html">get_study_info</a></span><span class="op">(</span><span class="st">"blca_nmibc_2017"</span><span class="op">)</span><span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span> 
<span class="co">#&gt;                             [,1]                                                                           </span>
<span class="co">#&gt; name                        "Nonmuscle Invasive Bladder Cancer (MSK Eur Urol 2017)"                        </span>
<span class="co">#&gt; description                 "IMPACT sequencing of 105 High Risk Nonmuscle Invasive Bladder Cancer samples."</span>
<span class="co">#&gt; publicStudy                 "TRUE"                                                                         </span>
<span class="co">#&gt; pmid                        "28583311"                                                                     </span>
<span class="co">#&gt; citation                    "Pietzak et al. Eur Urol 2017"                                                 </span>
<span class="co">#&gt; groups                      "PUBLIC"                                                                       </span>
<span class="co">#&gt; status                      "0"                                                                            </span>
<span class="co">#&gt; importDate                  "2022-03-04 17:49:56"                                                          </span>
<span class="co">#&gt; allSampleCount              "105"                                                                          </span>
<span class="co">#&gt; sequencedSampleCount        "105"                                                                          </span>
<span class="co">#&gt; cnaSampleCount              "105"                                                                          </span>
<span class="co">#&gt; mrnaRnaSeqSampleCount       "0"                                                                            </span>
<span class="co">#&gt; mrnaRnaSeqV2SampleCount     "0"                                                                            </span>
<span class="co">#&gt; mrnaMicroarraySampleCount   "0"                                                                            </span>
<span class="co">#&gt; miRnaSampleCount            "0"                                                                            </span>
<span class="co">#&gt; methylationHm27SampleCount  "0"                                                                            </span>
<span class="co">#&gt; rppaSampleCount             "0"                                                                            </span>
<span class="co">#&gt; massSpectrometrySampleCount "0"                                                                            </span>
<span class="co">#&gt; completeSampleCount         "0"                                                                            </span>
<span class="co">#&gt; readPermission              "TRUE"                                                                         </span>
<span class="co">#&gt; studyId                     "blca_nmibc_2017"                                                              </span>
<span class="co">#&gt; cancerTypeId                "blca"                                                                         </span>
<span class="co">#&gt; cancerType.name             "Bladder Urothelial Carcinoma"                                                 </span>
<span class="co">#&gt; cancerType.dedicatedColor   "Yellow"                                                                       </span>
<span class="co">#&gt; cancerType.shortName        "BLCA"                                                                         </span>
<span class="co">#&gt; cancerType.parent           "bladder"                                                                      </span>
<span class="co">#&gt; cancerType.cancerTypeId     "blca"                                                                         </span>
<span class="co">#&gt; referenceGenome             "hg19"</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/get_study_info.html">get_study_info</a></span><span class="op">(</span><span class="st">"prad_msk_2019"</span><span class="op">)</span><span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="op">)</span> 
<span class="co">#&gt;                             [,1]                                                             </span>
<span class="co">#&gt; name                        "Prostate Cancer (MSK, Cell Metab 2020)"                         </span>
<span class="co">#&gt; description                 "MSK-IMPACT Sequencing of 18 prostate cancer tumor/normal pairs."</span>
<span class="co">#&gt; publicStudy                 "TRUE"                                                           </span>
<span class="co">#&gt; pmid                        "31564440"                                                       </span>
<span class="co">#&gt; citation                    "Granlund et al. Cell Metab 2020"                                </span>
<span class="co">#&gt; groups                      "PUBLIC"                                                         </span>
<span class="co">#&gt; status                      "0"                                                              </span>
<span class="co">#&gt; importDate                  "2022-03-08 18:48:08"                                            </span>
<span class="co">#&gt; allSampleCount              "18"                                                             </span>
<span class="co">#&gt; sequencedSampleCount        "18"                                                             </span>
<span class="co">#&gt; cnaSampleCount              "18"                                                             </span>
<span class="co">#&gt; mrnaRnaSeqSampleCount       "0"                                                              </span>
<span class="co">#&gt; mrnaRnaSeqV2SampleCount     "0"                                                              </span>
<span class="co">#&gt; mrnaMicroarraySampleCount   "0"                                                              </span>
<span class="co">#&gt; miRnaSampleCount            "0"                                                              </span>
<span class="co">#&gt; methylationHm27SampleCount  "0"                                                              </span>
<span class="co">#&gt; rppaSampleCount             "0"                                                              </span>
<span class="co">#&gt; massSpectrometrySampleCount "0"                                                              </span>
<span class="co">#&gt; completeSampleCount         "0"                                                              </span>
<span class="co">#&gt; readPermission              "TRUE"                                                           </span>
<span class="co">#&gt; studyId                     "prad_msk_2019"                                                  </span>
<span class="co">#&gt; cancerTypeId                "prostate"                                                       </span>
<span class="co">#&gt; cancerType.name             "Prostate"                                                       </span>
<span class="co">#&gt; cancerType.dedicatedColor   "Cyan"                                                           </span>
<span class="co">#&gt; cancerType.shortName        "PROSTATE"                                                       </span>
<span class="co">#&gt; cancerType.parent           "tissue"                                                         </span>
<span class="co">#&gt; cancerType.cancerTypeId     "prostate"                                                       </span>
<span class="co">#&gt; referenceGenome             "hg19"</span></code></pre></div>
<p>Lastly, it is important to know what genomic data is available for our study. Not all studies in your database will have data available on all types of genomic information. For example, it is common for studies not to provide data on fusions.</p>
<p>We can check available genomic data with <code><a href="../reference/available_profiles.html">available_profiles()</a></code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/available_profiles.html">available_profiles</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 8</span></span>
<span class="co">#&gt;   molecularAlterationT… datatype name  description showProfileInAn… patientLevel</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> COPY_NUMBER_ALTERATI… DISCRETE Puta… Copy Numbe… TRUE             FALSE       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> MUTATION_EXTENDED     MAF      Muta… Mutation d… TRUE             FALSE       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> STRUCTURAL_VARIANT    FUSION   Fusi… Fusions.    TRUE             FALSE       </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 2 more variables: molecularProfileId &lt;chr&gt;, studyId &lt;chr&gt;</span></span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/available_profiles.html">available_profiles</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"prad_msk_2019"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 8</span></span>
<span class="co">#&gt;   molecularAlterationT… datatype name  description showProfileInAn… patientLevel</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> COPY_NUMBER_ALTERATI… DISCRETE Puta… Putative c… TRUE             FALSE       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> MUTATION_EXTENDED     MAF      Muta… IMPACT468 … TRUE             FALSE       </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> STRUCTURAL_VARIANT    FUSION   Fusi… Fusion dat… TRUE             FALSE       </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 2 more variables: molecularProfileId &lt;chr&gt;, studyId &lt;chr&gt;</span></span></code></pre></div>
<p>Luckily, our studies have mutation, copy number alteration and fusion data available. Each of these data types has a unique molecular profile ID. The molecular profile ID usually takes the form of <code>&lt;study_id&gt;_mutations</code>, <code>&lt;study_id&gt;_fusion</code>, <code>&lt;study_id&gt;_cna</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/available_profiles.html">available_profiles</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">molecularProfileId</span><span class="op">)</span>
<span class="co">#&gt; [1] "blca_nmibc_2017_cna"       "blca_nmibc_2017_mutations"</span>
<span class="co">#&gt; [3] "blca_nmibc_2017_fusion"</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="pulling-genomic-data">Pulling Genomic Data<a class="anchor" aria-label="anchor" href="#pulling-genomic-data"></a>
</h2>
<p>Now that we have inspected our studies and confirmed the genomic data that is available, we will pull the data into our R environment. We will show two ways to do this:</p>
<ol style="list-style-type: decimal">
<li>Using study IDs (<code><a href="../reference/get_genetics_by_study.html">get_genetics_by_study()</a></code>)</li>
<li>Using sample ID-study ID pairs (<code><a href="../reference/get_genetics_by_sample.html">get_genetics_by_sample()</a></code>)</li>
</ol>
<p>Pulling by study will give us genomic data for all genes/panels included in the study. You must pull by one study at a time. A benefit of pulling data this way is that you don’t have to specify which specific genes to query. It returns all genomic data available by default, however pulling all study data can be slow and inconvenient if you are working across multiple large studies, especially if you only need a subset of each study.</p>
<p>When we pull by sample IDs, the API requires us to specify which genes we want to query. If you do not specify, the function will default to querying all IMPACT genes (on any version of IMPACT panel). This allows you to pull specific samples across multiple studies.</p>
<div class="section level3">
<h3 id="by-study-ids">By Study IDs<a class="anchor" aria-label="anchor" href="#by-study-ids"></a>
</h3>
<p>To pull by study ID, we can pull each data type individually. Specifying which genes to query is not required when pulling data by study ID.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="va">mut_blca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mutations_by_study.html">get_mutations_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_mutations"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>
<span class="va">cna_blca</span><span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_cna_by_study.html">get_cna_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_cna"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>
<span class="va">fus_blca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_fusions_by_study.html">get_fusions_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_fusion"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>


<span class="va">mut_prad</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mutations_by_study.html">get_mutations_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"prad_msk_2019"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"prad_msk_2019_mutations"</span> molecular profile in the <span style="color: #0000BB;">"prad_msk_2019"</span> study</span>
<span class="va">cna_prad</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_cna_by_study.html">get_cna_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"prad_msk_2019"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"prad_msk_2019_cna"</span> molecular profile in the <span style="color: #0000BB;">"prad_msk_2019"</span> study</span>
<span class="va">fus_prad</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_fusions_by_study.html">get_fusions_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"prad_msk_2019"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"prad_msk_2019_fusion"</span> molecular profile in the <span style="color: #0000BB;">"prad_msk_2019"</span> study</span></code></pre></div>
<p>Or we can pull all genomic data at the same time with <code><a href="../reference/get_genetics_by_study.html">get_genetics_by_study()</a></code></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_genomic_blca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genetics_by_study.html">get_genetics_by_study</a></span><span class="op">(</span><span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_mutations"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_cna"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"blca_nmibc_2017_fusion"</span> molecular profile in the <span style="color: #0000BB;">"blca_nmibc_2017"</span> study</span>
<span class="va">all_genomic_prad</span><span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genetics_by_study.html">get_genetics_by_study</a></span><span class="op">(</span><span class="st">"prad_msk_2019"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"prad_msk_2019_mutations"</span> molecular profile in the <span style="color: #0000BB;">"prad_msk_2019"</span> study</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"prad_msk_2019_cna"</span> molecular profile in the <span style="color: #0000BB;">"prad_msk_2019"</span> study</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"prad_msk_2019_fusion"</span> molecular profile in the <span style="color: #0000BB;">"prad_msk_2019"</span> study</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://dplyr.tidyverse.org/reference/all_equal.html" class="external-link">all_equal</a></span><span class="op">(</span><span class="va">mut_blca</span>, <span class="va">all_genomic_blca</span><span class="op">$</span><span class="va">mutation</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/all_equal.html" class="external-link">all_equal</a></span><span class="op">(</span><span class="va">cna_blca</span>, <span class="va">all_genomic_blca</span><span class="op">$</span><span class="va">cna</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="fu"><a href="https://dplyr.tidyverse.org/reference/all_equal.html" class="external-link">all_equal</a></span><span class="op">(</span><span class="va">fus_blca</span>, <span class="va">all_genomic_blca</span><span class="op">$</span><span class="va">fusion</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#get_panel_by_sample</span></code></pre></div>
<p>Finally, we can join the two studies together</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mut_study</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">mut_blca</span>, <span class="va">mut_prad</span><span class="op">)</span>
<span class="va">cna_study</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">cna_blca</span>, <span class="va">cna_prad</span><span class="op">)</span>
<span class="va">fus_study</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">fus_blca</span>, <span class="va">fus_prad</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="by-sample-ids">By Sample IDs<a class="anchor" aria-label="anchor" href="#by-sample-ids"></a>
</h3>
<p>When we pull by sample IDs, the API requires us to specify which genes we want to query. If you do not specify, the function will default to querying all IMPACT genes (on any version of IMPACT panel). This allows you to pull specific samples across multiple studies.</p>
<p>We can pull from both studies at the same time for a select set of samples using the <code>sample_study_pairs</code> argument in <code><a href="../reference/get_genetics_by_sample.html">get_genetics_by_sample()</a></code>.</p>
<p>Let’s pull data for the first 10 samples in each study. We first need to construct our dataframe to pass to the function:</p>
<p><em>Note: you can also run <code><a href="../reference/available_patients.html">available_patients()</a></code> to only pull patient IDs</em></p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">s1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_samples.html">available_samples</a></span><span class="op">(</span><span class="st">"blca_nmibc_2017"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sampleId</span>, <span class="va">patientId</span>, <span class="va">studyId</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>

<span class="va">s2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_samples.html">available_samples</a></span><span class="op">(</span><span class="st">"prad_msk_2019"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sampleId</span>,  <span class="va">patientId</span>, <span class="va">studyId</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">10</span><span class="op">)</span>

<span class="va">df_pairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">s1</span>, <span class="va">s2</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">patientId</span><span class="op">)</span></code></pre></div>
<p>We need to rename the columns as per the functions documentation.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df_pairs</span> <span class="op">&lt;-</span> <span class="va">df_pairs</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="st">"sample_id"</span> <span class="op">=</span> <span class="va">sampleId</span>, 
         <span class="st">"study_id"</span> <span class="op">=</span> <span class="va">studyId</span><span class="op">)</span></code></pre></div>
<p>Now we pass this to <code><a href="../reference/get_genetics_by_sample.html">get_genetics_by_sample()</a></code></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_genomic</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genetics_by_sample.html">get_genetics_by_sample</a></span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs</span><span class="op">)</span>
<span class="co">#&gt; Joining, by = "study_id"</span>
<span class="co">#&gt; The following parameters were used in query:</span>
<span class="co">#&gt; <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_nmibc_2017"</span> and <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">blca_nmibc_2017_mutations</span> and <span style="color: #0000BB;">prad_msk_2019_mutations</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"all IMPACT genes (see `gnomeR::impact_gene_info`)"</span></span>
<span class="co">#&gt; Joining, by = "study_id"</span>
<span class="co">#&gt; The following parameters were used in query:</span>
<span class="co">#&gt; <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_nmibc_2017"</span> and <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">blca_nmibc_2017_cna</span> and <span style="color: #0000BB;">prad_msk_2019_cna</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"all IMPACT genes (see `gnomeR::impact_gene_info`)"</span></span>
<span class="co">#&gt; Joining, by = "study_id"</span>
<span class="co">#&gt; The following parameters were used in query:</span>
<span class="co">#&gt; <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_nmibc_2017"</span> and <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">blca_nmibc_2017_fusion</span> and <span style="color: #0000BB;">prad_msk_2019_fusion</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"all IMPACT genes (see `gnomeR::impact_gene_info`)"</span></span>

<span class="va">mut_sample</span> <span class="op">&lt;-</span> <span class="va">all_genomic</span><span class="op">$</span><span class="va">mutation</span></code></pre></div>
<p>Like with querying by study ID, you can also pull data individually by genomic data type:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">mut_only</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mutations_by_sample.html">get_mutations_by_sample</a></span><span class="op">(</span>sample_study_pairs <span class="op">=</span> <span class="va">df_pairs</span><span class="op">)</span>
<span class="co">#&gt; Joining, by = "study_id"</span>
<span class="co">#&gt; The following parameters were used in query:</span>
<span class="co">#&gt; <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"blca_nmibc_2017"</span> and <span style="color: #0000BB;">"prad_msk_2019"</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">blca_nmibc_2017_mutations</span> and <span style="color: #0000BB;">prad_msk_2019_mutations</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"all IMPACT genes (see `gnomeR::impact_gene_info`)"</span></span>

<span class="fu"><a href="https://rdrr.io/r/base/identical.html" class="external-link">identical</a></span><span class="op">(</span><span class="va">mut_only</span>, <span class="va">mut_sample</span><span class="op">)</span>
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="compare-pulling-by-study-vs--sample">Compare Pulling by Study vs. Sample<a class="anchor" aria-label="anchor" href="#compare-pulling-by-study-vs--sample"></a>
</h3>
<p>Let’s compare these results with the ones we got from pulling by study:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co"># filter to our subset used in sample query</span>
<span class="va">mut_study_subset</span> <span class="op">&lt;-</span> <span class="va">mut_study</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sampleId</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span>  <span class="va">df_pairs</span><span class="op">$</span><span class="va">sample_id</span><span class="op">)</span>

<span class="co"># arrange to compare</span>
<span class="va">mut_study_subset</span> <span class="op">&lt;-</span> <span class="va">mut_study_subset</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">sampleId</span><span class="op">)</span><span class="op">)</span><span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">entrezGeneId</span><span class="op">)</span><span class="op">)</span>

<span class="va">mut_sample</span> <span class="op">&lt;-</span> <span class="va">mut_sample</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">sampleId</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">entrezGeneId</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/base/all.equal.html" class="external-link">all.equal</a></span><span class="op">(</span><span class="va">mut_study_subset</span>, <span class="va">mut_sample</span><span class="op">)</span>
<span class="co">#&gt; [1] "Names: 6 string mismatches"                                     </span>
<span class="co">#&gt; [2] "Component 28: 'is.NA' value mismatch: 0 in current 10 in target"</span>
<span class="co">#&gt; [3] "Component 29: 176 string mismatches"                            </span>
<span class="co">#&gt; [4] "Component 30: 176 string mismatches"                            </span>
<span class="co">#&gt; [5] "Component 31: Modes: character, numeric"                        </span>
<span class="co">#&gt; [6] "Component 31: target is character, current is numeric"          </span>
<span class="co">#&gt; [7] "Component 32: Mean relative difference: 0.003716705"            </span>
<span class="co">#&gt; [8] "Component 33: Modes: numeric, character"                        </span>
<span class="co">#&gt; [9] "Component 33: target is numeric, current is character"</span></code></pre></div>
<p>In this case, both results are equal which means both studies were IMPACT panel only studies.</p>
<p>Below is an example of when the results may not be equal. This occurs if one of the studies used a wider panel, because we used the default gene set of IMPACT only genes when we searched by sample:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ex_by_sample</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/get_mutations_by_sample.html">get_mutations_by_sample</a></span><span class="op">(</span>sample_id <span class="op">=</span> <span class="st">"PD3181a"</span>,
                                        study_id <span class="op">=</span> <span class="st">"acyc_sanger_2013"</span><span class="op">)</span>
<span class="co">#&gt; The following parameters were used in query:</span>
<span class="co">#&gt; <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"acyc_sanger_2013"</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">"acyc_sanger_2013_mutations"</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">"all IMPACT genes (see `gnomeR::impact_gene_info`)"</span></span>

<span class="va">ex_by_study</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_mutations_by_study.html">get_mutations_by_study</a></span><span class="op">(</span>study_id <span class="op">=</span> <span class="st">"acyc_sanger_2013"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sampleId</span> <span class="op">==</span> <span class="st">"PD3181a"</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"acyc_sanger_2013_mutations"</span> molecular profile in the <span style="color: #0000BB;">"acyc_sanger_2013"</span> study</span>

<span class="va">genes_diff</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">ex_by_study</span><span class="op">$</span><span class="va">entrezGeneId</span>, <span class="va">ex_by_sample</span><span class="op">$</span><span class="va">entrezGeneId</span><span class="op">)</span></code></pre></div>
<p>We could resolve this discrepancy by specifying which genes we want to query in <code><a href="../reference/get_mutations_by_sample.html">get_mutations_by_sample()</a></code> to ensure that the same genes are queried across both studies:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ex_by_sample_update</span> <span class="op">&lt;-</span>  <span class="fu"><a href="../reference/get_mutations_by_sample.html">get_mutations_by_sample</a></span><span class="op">(</span>sample_id <span class="op">=</span> <span class="st">"PD3181a"</span>,
                                               study_id <span class="op">=</span> <span class="st">"acyc_sanger_2013"</span>, 
                                               genes <span class="op">=</span> <span class="va">ex_by_study</span><span class="op">$</span><span class="va">entrezGeneId</span><span class="op">)</span>
<span class="co">#&gt; The following parameters were used in query:</span>
<span class="co">#&gt; <span style="color: #00BB00;">Study ID</span>: <span style="color: #0000BB;">"acyc_sanger_2013"</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Molecular Profile ID</span>: <span style="color: #0000BB;">"acyc_sanger_2013_mutations"</span></span>
<span class="co">#&gt; <span style="color: #00BB00;">Genes</span>: <span style="color: #0000BB;">747</span>, <span style="color: #0000BB;">2318</span>, <span style="color: #0000BB;">2639</span>, <span style="color: #0000BB;">4360</span>, <span style="color: #0000BB;">4654</span>, <span style="color: #0000BB;">4969</span>, <span style="color: #0000BB;">5188</span>, <span style="color: #0000BB;">6710</span>, <span style="color: #0000BB;">6814</span>, <span style="color: #0000BB;">8289</span>, <span style="color: #0000BB;">10556</span>, <span style="color: #0000BB;">26266</span>,</span>
<span class="co">#&gt; <span style="color: #0000BB;">27130</span>, <span style="color: #0000BB;">79613</span>, <span style="color: #0000BB;">81794</span>, <span style="color: #0000BB;">93643</span>, <span style="color: #0000BB;">113655</span>, <span style="color: #0000BB;">114112</span>, <span style="color: #0000BB;">114821</span>, <span style="color: #0000BB;">118491</span>, <span style="color: #0000BB;">118491</span>, <span style="color: #0000BB;">127435</span>,</span>
<span class="co">#&gt; <span style="color: #0000BB;">127731</span>, <span style="color: #0000BB;">152485</span>, <span style="color: #0000BB;">284525</span>, and <span style="color: #0000BB;">401024</span></span>

<span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">ex_by_study</span><span class="op">$</span><span class="va">entrezGeneId</span>, <span class="va">ex_by_sample_update</span><span class="op">$</span><span class="va">entrezGeneId</span><span class="op">)</span>
<span class="co">#&gt; integer(0)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="pulling-clinical-data">Pulling Clinical Data<a class="anchor" aria-label="anchor" href="#pulling-clinical-data"></a>
</h2>
<p>You can also pull clinical data by study ID, sample ID, or patient ID. Pulling by sample ID will pull all sample-level characteristics (e.g. sample site, tumor stage at sampling time and other variables collected at time of sampling that may be available). Pulling by patient ID will pull all patient-level characteristics (e.g. age, sex, etc.). Pulling by study ID will pull all sample <em>and</em> patient-level characteristics at once.</p>
<p>You can explore what clinical data is available a study using:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">attr_blca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_clinical_attributes.html">available_clinical_attributes</a></span><span class="op">(</span><span class="st">"blca_nmibc_2017"</span><span class="op">)</span>
<span class="va">attr_prad</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_clinical_attributes.html">available_clinical_attributes</a></span><span class="op">(</span><span class="st">"prad_msk_2019"</span><span class="op">)</span>

<span class="va">attr_prad</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 13 × 7</span></span>
<span class="co">#&gt;    displayName   description datatype patientAttribute priority clinicalAttribu…</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Cancer Type   Cancer Type STRING   FALSE            1        CANCER_TYPE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Cancer Type … Cancer Typ… STRING   FALSE            1        CANCER_TYPE_DET…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Fraction Gen… Fraction G… NUMBER   FALSE            20       FRACTION_GENOME…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Gene Panel    Gene Panel. STRING   FALSE            1        GENE_PANEL      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Mutation Cou… Mutation C… NUMBER   FALSE            30       MUTATION_COUNT  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Oncotree Code Oncotree C… STRING   FALSE            1        ONCOTREE_CODE   </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Sample Class  The sample… STRING   FALSE            1        SAMPLE_CLASS    </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Number of Sa… Number of … STRING   TRUE             1        SAMPLE_COUNT    </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Sample Type   The type o… STRING   FALSE            1        SAMPLE_TYPE     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Sex           Sex         STRING   TRUE             1        SEX             </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Somatic Stat… Somatic St… STRING   FALSE            1        SOMATIC_STATUS  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">12</span> Specimen Pre… The method… STRING   FALSE            1        SPECIMEN_PRESER…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">13</span> TMB (nonsyno… TMB (nonsy… NUMBER   FALSE            1        TMB_NONSYNONYMO…</span>
<span class="co">#&gt; <span style="color: #949494;"># … with 1 more variable: studyId &lt;chr&gt;</span></span></code></pre></div>
<p>There are a select set available for both studies:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">in_both</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">attr_blca</span><span class="op">$</span><span class="va">clinicalAttributeId</span>, <span class="va">attr_prad</span><span class="op">$</span><span class="va">clinicalAttributeId</span><span class="op">)</span></code></pre></div>
<p>The below pulls data at the sample level:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">clinical_blca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_clinical_by_sample.html">get_clinical_by_sample</a></span><span class="op">(</span>sample_id <span class="op">=</span> <span class="va">s1</span><span class="op">$</span><span class="va">sampleId</span>,
                       study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span>, 
                       clinical_attribute <span class="op">=</span> <span class="va">in_both</span><span class="op">)</span>

<span class="va">clinical_prad</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_clinical_by_sample.html">get_clinical_by_sample</a></span><span class="op">(</span>sample_id <span class="op">=</span> <span class="va">s2</span><span class="op">$</span><span class="va">sampleId</span>,
                       study_id <span class="op">=</span> <span class="st">"prad_msk_2019"</span>, 
                       clinical_attribute <span class="op">=</span> <span class="va">in_both</span><span class="op">)</span>

<span class="va">all_clinical</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">clinical_blca</span>, <span class="va">clinical_prad</span><span class="op">)</span>

<span class="va">all_clinical</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">"unique"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 5</span></span>
<span class="co">#&gt;   sampleId          patientId studyId         clinicalAttributeId     value     </span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 CANCER_TYPE             Bladder C…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 CANCER_TYPE_DETAILED    Bladder U…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 FRACTION_GENOME_ALTERED 0.4448    </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 MUTATION_COUNT          11        </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 ONCOTREE_CODE           BLCA      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">6</span> P-0001453-T01-IM3 P-0001453 blca_nmibc_2017 SOMATIC_STATUS          Matched</span></code></pre></div>
<p>The below pulls data at the patient level:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">p1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/available_patients.html">available_patients</a></span><span class="op">(</span><span class="st">"blca_nmibc_2017"</span><span class="op">)</span>

<span class="va">clinical_blca</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_clinical_by_patient.html">get_clinical_by_patient</a></span><span class="op">(</span>patient_id <span class="op">=</span> <span class="va">s1</span><span class="op">$</span><span class="va">patientId</span>,
                       study_id <span class="op">=</span> <span class="st">"blca_nmibc_2017"</span>, 
                       clinical_attribute <span class="op">=</span> <span class="va">in_both</span><span class="op">)</span>

<span class="va">clinical_prad</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_clinical_by_sample.html">get_clinical_by_sample</a></span><span class="op">(</span>sample_id <span class="op">=</span> <span class="va">s2</span><span class="op">$</span><span class="va">patientId</span>,
                       study_id <span class="op">=</span> <span class="st">"prad_msk_2019"</span>, 
                       clinical_attribute <span class="op">=</span> <span class="va">in_both</span><span class="op">)</span>

<span class="va">all_clinical</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">clinical_blca</span>, <span class="va">clinical_prad</span><span class="op">)</span>

<span class="va">all_clinical</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">"unique"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span>
<span class="co">#&gt;   patientId studyId         clinicalAttributeId value</span>
<span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0001453 blca_nmibc_2017 SAMPLE_COUNT        1    </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0001453 blca_nmibc_2017 SEX                 Male </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0002166 blca_nmibc_2017 SAMPLE_COUNT        1    </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0002166 blca_nmibc_2017 SEX                 Male </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">5</span> P-0003238 blca_nmibc_2017 SAMPLE_COUNT        1    </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">6</span> P-0003238 blca_nmibc_2017 SEX                 Male</span></code></pre></div>
<p>Like with the genomic data pull functions, you can also pull clinical data by study ID - sampleID, or ID study ID - patient ID pairs:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df_pairs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="va">s1</span>, <span class="va">s2</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="va">sampleId</span><span class="op">)</span>

<span class="va">df_pairs</span> <span class="op">&lt;-</span> <span class="va">df_pairs</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span><span class="st">"patient_id"</span> <span class="op">=</span> <span class="va">patientId</span>, 
         <span class="st">"study_id"</span> <span class="op">=</span> <span class="va">studyId</span><span class="op">)</span></code></pre></div>
<p>Now we pass this to <code><a href="../reference/get_genetics_by_sample.html">get_genetics_by_sample()</a></code></p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">all_patient_clinical</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_clinical_by_patient.html">get_clinical_by_patient</a></span><span class="op">(</span>patient_study_pairs <span class="op">=</span> <span class="va">df_pairs</span>, 
                                                clinical_attribute <span class="op">=</span> <span class="va">in_both</span><span class="op">)</span>

<span class="va">all_patient_clinical</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="op">-</span><span class="fu"><a href="https://tidyselect.r-lib.org/reference/starts_with.html" class="external-link">contains</a></span><span class="op">(</span><span class="st">"unique"</span><span class="op">)</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 34 × 4</span></span>
<span class="co">#&gt;    patientId studyId         clinicalAttributeId value </span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0001453 blca_nmibc_2017 SAMPLE_COUNT        1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0001453 blca_nmibc_2017 SEX                 Male  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0002166 blca_nmibc_2017 SAMPLE_COUNT        1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0002166 blca_nmibc_2017 SEX                 Male  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0003238 blca_nmibc_2017 SAMPLE_COUNT        1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0003238 blca_nmibc_2017 SEX                 Male  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0003257 blca_nmibc_2017 SAMPLE_COUNT        1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0003257 blca_nmibc_2017 SEX                 Female</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0003261 blca_nmibc_2017 SAMPLE_COUNT        1     </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0003261 blca_nmibc_2017 SEX                 Male  </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 24 more rows</span></span></code></pre></div>
<p>```</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Karissa Whiting.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.3.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
